fileServer.js

```diff
@@ -17,96 +19,69 @@
 
  function send404Request(res) {
    fs.readFile('public/404.html', 'UTF-8', function(err, html) {
-<<<<<<< HEAD
      if (err) {
        res.writeHead(500);
-       res.write("Internal Server Error");
+       res.write('Internal Server Error');
+       console.log(err);
      } else {
        res.writeHead(404, {
-         'Content-type': mimeType
+         'Content-type': 'text/html'
        });
        res.end(html);
      }
-=======
-     res.setHeader('Content-type', mimeType);
-     res.end(html);
->>>>>>> 945100f8f04e33242f280c0f03422c5c690fe44b
    });
  }
 
  function onRequest(req, res) {
-<<<<<<< HEAD
-   console.log('request starting....');
-=======
-   console.log('request starting...');
->>>>>>> 945100f8f04e33242f280c0f03422c5c690fe44b
-
-   var filename;
-   var pathname = url.parse(req.url).pathname;
-   filename = path.join(rootDir, pathname);
-   console.log('Filename: ' + filename);
-<<<<<<< HEAD
+   console.log(`${req.method} ${req.url}`);
+
+   var pathname = url.parse(req.url).path;
+
+   
+   var filename = path.join(rootDir, pathname);
+
+   var headers = {'Content-type': String(mimeType[path.extname 
+         (filename)])};
+
    var stream = fs.createReadStream(filename);
-=======
->>>>>>> 945100f8f04e33242f280c0f03422c5c690fe44b
 
    if (req.method !== 'GET') {
      res.writeHead(405, {
-       "Content-type": "text/plain"
+       "Content-type": 'text/plain'
      });
      res.write(`HTTP method ${req.method} not yet supported`);
      res.end();
-     return
+     return;
    }
 
    if (req.url === '/') {
-<<<<<<< HEAD
      fs.readFile('public/index.html', 'UTF-8', function(err, html) {
+
        if (err) {
          res.writeHead(500);
-         res.write("Internal Server Error");
+         res.write('Internal Server Error');
```

Comment:
Same as 500 above - return the real error to the user. Your refactoring alarm bell should go off, as you have almost the same code for handling 500 errors in two places, which means ... you got it, the 500 error generation code should be a function that takes the error message as an argument e.g. ```generate500error(errorMessage)```